src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -0,0 +1,321 @@
+            // If there is an annotation, the javadoc comment will be a child of it.
+            if (child.getType() == TokenTypes.ANNOTATION) {
+                firstSentence = getFirstJavadocSentence(child);
+            }
+            // Otherwise, the javadoc comment will be right here.
+            else if (child.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
+                    && JavadocUtils.isJavadocComment(child)) {
+                final DetailNode tree = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(child);
+                firstSentence = getFirstJavadocSentence(tree);
+            }
+        }
+        return firstSentence;
+    }
+
+    /**
+     * Extracts the first sentence of the comment of an DetailAST.
+     * @param tree to extract the first sentence
+     * @return the first sentence of the node or {@code null} if the first sentence is absent or
+     *      malformed (does not end with period)
+     * @throws CheckstyleException if an unsupported inline tag found
+     */
+    private static String getFirstJavadocSentence(DetailNode tree) throws CheckstyleException {
+        String firstSentence = null;
+        final StringBuilder builder = new StringBuilder(128);
+        for (DetailNode node : tree.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.TEXT) {
+                final Matcher matcher = END_OF_SENTENCE_PATTERN.matcher(node.getText());
+                if (matcher.find()) {
+                    // Commit the sentence if the end-of-sentence marker is found.
+                    firstSentence = builder.append(matcher.group(1)).toString();
+                    break;
+                }
+                // Otherwise append the whole line and look for the end-of-sentence marker
+                // on the next line.
+                builder.append(node.getText());
+            }
+            else if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                formatInlineCodeTag(builder, node);
+            }
+            else if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                formatHtmlElement(builder, node);
+            }
+        }
+        return firstSentence;
+    }
+
+    /**
+     * Converts inline code tag into HTML form.
+     * @param builder to append
+     * @param inlineTag to format
+     * @throws CheckstyleException if the inline javadoc tag is not a literal nor a code tag
+     */
+    private static void formatInlineCodeTag(StringBuilder builder, DetailNode inlineTag)
+            throws CheckstyleException {
+        boolean isCodeTag = false;
+        for (DetailNode node : inlineTag.getChildren()) {
+            switch (node.getType()) {
+                // Ignore opening and closing curly braces and spaces after the tag.
+                case JavadocTokenTypes.JAVADOC_INLINE_TAG_START:
+                case JavadocTokenTypes.JAVADOC_INLINE_TAG_END:
+                case JavadocTokenTypes.WS:
+                    break;
```

Comment:
all breaks should be on the bottom. Stuff to ignore human should read the last.

Reply:
done